src/lib/utils/errors.ts

```diff
@@ -15,14 +15,24 @@ export function toReadableError(errorText: string, error: unknown) {
   return new Error(`${errorText} 👺 ${error}`);
 }
 
-export function extractUserFriendlyError(error: Error): string | undefined {
+export function extractUserFriendlyError(
+  error: unknown,
+  {
+    defaultError = "Please retry this action"
+  }: {
+    defaultError?: string;
+  } = {}
+): string {
+  if (!error || typeof error !== "object") {
+    return defaultError;
+  }
   const m = error.toString().match(/(?<=execution reverted: ).*/)?.[0];
```

Comment:
regex extraction is not giving expected results here
![image](https://github.com/bosonprotocol/interface/assets/7184124/9fa3a704-a2b4-450c-9d54-a6e007145a0b)

Reply:
However, it's working with meta-transactions because the error is formatted differently (see screenshot).
Maybe useful to support both cases
![image](https://github.com/bosonprotocol/interface/assets/7184124/d7c63f8f-ad57-4dd1-bec9-9087552e1cf2)